kite-data/kite-data-core/src/main/java/org/kitesdk/data/View.java

```diff
@@ -128,4 +128,41 @@
    * @since 0.16.0
    */
   public URI getUri();
+
+  /**
+   * Signal that the view is ready for consumption.
+   *
+   * May result in a no-op if the view implementation does not support the ready
+   * concept.
+   *
+   * Note that an {@link #isEmpty() empty} view may be signaled as ready.
+   *
+   * @throws UnsupportedOperationException if the view does not support the ready
+   *          concept.
```

Comment:
We went back and forth on using UnsupportedOperationException here, @ben-roling's previous pull around this concept (#100) started in this direction and worked toward another interface and custom exception for the ready concept. UnsupportedOperationException seems to more closely match the pattern established in the "deleteAll" API so I stuck with that for the initial pass, and signals feel like they could be widely supported across dataset implementations.

Now that more time has passed since that original pull request I wanted to get some thoughts on if this feels like the better direction or adding a layered interface feels more appropriate.